src/WebJobs.Script.WebHost/Middleware/VirtualFileSystemMiddleware.cs

```diff
@@ -0,0 +1,119 @@
+        {
+            return context.Request.Path.StartsWithSegments("/admin/vfs") ||
+                context.Request.Path.StartsWithSegments("/admin/zip");
+        }
+
+        public async Task InvokeAsync(HttpContext context, RequestDelegate _)
+        {
+            var authorized = await AuthenticateAndAuthorize(context);
+
+            if (!authorized)
+            {
+                context.Response.StatusCode = StatusCodes.Status401Unauthorized;
+            }
+            else
+            {
+                await InternalInvokeAsync(context);
+            }
+        }
+
+        private async Task InternalInvokeAsync(HttpContext context)
+        {
+            // choose the right instance to use.
+            var handler = context.Request.Path.StartsWithSegments("/admin/vfs") ? _vfs : _zip;
+            HttpResponseMessage response = null;
+            try
+            {
+                switch (context.Request.Method.ToLowerInvariant())
+                {
+                    case "get":
+                        response = await handler.GetItem(context.Request);
+                        break;
+
+                    case "put":
+                        response = await handler.PutItem(context.Request);
+                        break;
+
+                    case "delete":
+                        response = await handler.DeleteItem(context.Request);
+                        break;
+
+                    default:
+                        // VFS only supports GET, PUT, and DELETE
+                        response = new HttpResponseMessage(System.Net.HttpStatusCode.MethodNotAllowed);
+                        break;
+                }
+
+                context.Response.StatusCode = (int)response.StatusCode;
+
+                // write response headers
+                context.Response.Headers.AddRange(response.Headers.ToCoreHeaders());
+
+                // This is to handle NullContent which != null, but has ContentLength of null.
+                if (response.Content != null && response.Content.Headers.ContentLength != null)
+                {
+                    // Exclude content length to let ASP.NET Core take care of setting that based on the stream size.
+                    context.Response.Headers.AddRange(response.Content.Headers.ToCoreHeaders("Content-Length"));
+                    await response.Content.CopyToAsync(context.Response.Body);
+                }
+                response.Dispose();
+            }
+            catch (Exception e)
+            {
+                if (response != null)
+                {
+                    response.Dispose();
+                }
+
+                await context.Response.WriteAsync(e.Message);
+            }
+        }
+
+        private async Task<bool> AuthenticateAndAuthorize(HttpContext context)
+        {
+            var authorizationEvaluator = context.RequestServices.GetRequiredService<IAuthorizationPolicyProvider>();
```

Comment:
nit: change variable name to `authorizationPolicyProvider` or `prolicyProvider`

Reply:
done